lib/galaxy/datatypes/sequence.py

```diff
@@ -940,4 +946,84 @@ def sniff(self, filename):
                             pairs = True
                     if seq and coor and pairs:
                         return True
+        return False
+
+
+class DotBracket ( Sequence ):
+    edam_format = "format_1457"
+    file_ext = "dbn"
+    
+    sequence_regexp = re.compile( "^[ACGTURYKMSWBDHVN]+$", re.I)
+    structure_regexp = re.compile( "^[\(\)\.\[\]]*" )
+    
+    def set_meta( self, dataset, **kwd ):
+        """
+        Set the number of sequences and the number of data lines
+        in dataset.
+        """
+        if self.max_optional_metadata_filesize >= 0 and dataset.get_size() > self.max_optional_metadata_filesize:
+            dataset.metadata.data_lines = None
+            dataset.metadata.sequences = None
+            dataset.metadata.seconday_structures = None
+            return
+        
+        data_lines = 0
+        sequences = 0
+        
+        for line in file( dataset.file_name ):
+            line = line.strip()
+            data_lines += 1
+            
+            if line and line.startswith( '>' ):
+                sequences += 1
+        
+        dataset.metadata.data_lines = data_lines
+        dataset.metadata.sequences = sequences
+    
+    def sniff(self, filename):
+        """
+        The Dot-Bracket format is as follows:
+        
+        >sequenceName1
+        CCCaaaGGG
+        (((...)))
+        >sequenceName2
+        GGGuuuCCC
+        (((...)))
+        
+        Because it remains unclear whether the Dot-Bracket format may
+        contain multiple sequences per file, sniffing is only applied
+        on the first 3 lines.
+        """
+        
+        i = 0
+        
+        with open( filename, "r" ) as handle:
+            for line in handle:
+                line = line.strip()
+                
+                state = i % 3
```

Comment:
@bgruening How do you feel about solving this by putting a crystal clear specification om the Galaxy wiki?
https://wiki.galaxyproject.org/Learn/Datatypes

I would propose something like:

== Dbn ==

The Dot-Bracket Notation format (```.dbn```) is a commonly used format to describe 2D RNA structures, in particular as output for structure predictors that can't predict pseudoknots.
Every entry consists of three lines:
* 1. The header line, to provide information about the sequence (e.g. name, free energy).
* 2. The sequence line, to provide the sequence of which the structures will be described.
* 3. The structure (Dot-Bracket) line, describing the structure of the sequence.
* Two nucleotides that form a bond are indicated with encloding parenthesis and dots represent unbound nucleotides.
* Pseudoknots are indicated with { .. } or [ .. ].

Example of a Dbn file:

```
>Sequence1 information - small hairpin (dG=-10.3)
ggggaaacccc
((((...))))
>Sequence2 information (dG=0.0)
aaaaaaaaaaa
...........
```

[[http://ultrastudio.org/en/Dot-Bracket_Notation]]

[[http://rna.urmc.rochester.edu/Text/File_Formats.html#DotBracket]]

==== Galaxy Dbn (Dot-Bracket notation) rules:====
* The first non-empty line is a header line: no comment lines are allowed.
* A header line starts with a '```>```' symbol and continues with 0 or multiple symbols until the line ends.
* The second non-empty line is a sequence line.
* A sequence line may only include chars that match the Fasta format ([[https://en.wikipedia.org/wiki/FASTA_format#Sequence_representation]]) symbols for nucleotides: ```ACGTURYKMSWBDHVN```, and may thus not include whitespaces.
* A sequence line has no prefix and no suffix.
* A sequence line is case insensitive.
* The third non-empty line is a structure (Dot-Bracket) line and only describes the 2D structure of the sequence above it.
* A structure line must consist of the following chars: '```.{}[]()```'.
* A structure line must be of the same length as the sequence line, and each char represents the structure of the nucleotide above it.
* A structure line has no prefix and no suffix.
* A nucleotide pairs with only 1 or 0 other nucleotides.
* In a structure line, the number of '```(```' symbols equals the number of '```)```' symbols, the number of '```[```' symbols equals the number of '```]```' symbols and the number of '```{```' symbols equals the number of '```}```' symbols.
* The format accepts multiple entries per file, given that each entry is provided as three lines: the header, sequence and structure line.
* Empty lines are allowed.